services/web/client/source/class/osparc/component/workbench/NodeUI.js

```diff
@@ -107,8 +107,28 @@ qx.Class.define("osparc.component.workbench.NodeUI", {
       return this.getNode().getNodeId();
     },
 
-    getMetaData: function() {
-      return this.getNode().getMetaData();
+    getCaptionBar: function() {
+      return this.getChildControl("captionbar");
+    },
+
+    setTriSelected: function(triState) {
```

Comment:
for this I see two cleaner options:
* selection is a new property and its value is binded to the backgrounColor/appearance properties
* use ``addState`` and ``removeState`` http://www.qooxdoo.org/devel/api/#qx.ui.core.Widget~addState!method_public and directly use them in the ``Appearance`` class.
Also a combination of both would do it nicely, and update the state inside the ``_applySelection`` method.

Reply:
Great. I have a couple of ideas in mind for this. Let's discuss it the coming days.